python/pyspark/sql/connect/table_arg.py

```diff
@@ -0,0 +1,101 @@
+#
+# Licensed to the Apache Software Foundation (ASF) under one or more
+# contributor license agreements.  See the NOTICE file distributed with
+# this work for additional information regarding copyright ownership.
+# The ASF licenses this file to You under the Apache License, Version 2.0
+# (the "License"); you may not use this file except in compliance with
+# the License.  You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+from typing import (
+    Iterable,
+    TYPE_CHECKING,
+    Union,
+    Sequence,
+    List,
+    Tuple,
+    cast,
+)
+
+import pyspark.sql.connect.proto as proto
+from pyspark.sql.column import Column
+from pyspark.sql.connect.expressions import SubqueryExpression, SortOrder
+from pyspark.sql.connect.functions import builtin as F
+from pyspark.sql.table_arg import TableArg as ParentTableArg
+
+from pyspark.errors import PySparkValueError
+
+if TYPE_CHECKING:
+    from pyspark.sql._typing import ColumnOrName
+    from pyspark.sql.connect.client import SparkConnectClient
+
+
+def _to_cols(cols: Tuple[Union["ColumnOrName", Sequence["ColumnOrName"]], ...]) -> List[Column]:
+    if len(cols) == 1 and isinstance(cols[0], list):
+        cols = cols[0]  # type: ignore[assignment]
+    return [F._to_col(c) for c in cast(Iterable["ColumnOrName"], cols)]
+
+
+class TableArg(ParentTableArg):
+    def __init__(self, subquery_expr: SubqueryExpression):
+        self._subquery_expr = subquery_expr
+
+    def _is_partitioned(self) -> bool:
+        """Checks if partitioning is already applied."""
+        return (
+            bool(self._subquery_expr._partition_spec) or self._subquery_expr._with_single_partition
+        )
+
+    def partitionBy(self, *cols: "ColumnOrName") -> "TableArg":
+        if self._is_partitioned():
+            raise PySparkValueError(
```

Comment:
`IllegalArgumentException` to be consistent with classic?

Reply:
Changed to pyspark.errors.IllegalArgumentException, that's good to know!